### PR TITLE
Add more to the HTTP/HTTPS tests

### PR DESCRIPTION
**Pull request by rjemanuele**
* test the stream end

### CODE REVIEW

File: modules/http.lua [hunk 1]
```diff
@@ -222,9 +223,21 @@ function ClientRequest:initialize(options, callback)
...
     end
+    local uafound = key:lower() == 'user-agent'
+    if uafound then
+      user_agent_found = value
+    end
     table.insert(self, header)
   end
 
+  if not user_agent then
+    user_agent = 'luvit/http/' .. exports.version .. ' luvi/' .. luvi.version
```
Comment (creationix) [inline modules/http.lua]: Should calculate the default value only once outside the closure.  It's not going to change for the lifetime of the process.

**Comment by creationix** [comment]
Rebase on luvi-up since http-lax has been merged?

**Comment by rjemanuele** [comment]
rebased

File: modules/http.lua [hunk 2]
```diff
@@ -207,11 +208,18 @@ end
 local ClientRequest = Writable:extend()
 exports.ClientRequest = ClientRequest
 
+function exports.ClientRequest.getDefaultUserAgent()
+  if exports.ClientRequest.default_user_agent ~= nil then
+    exports.ClientRequest.default_user_agent = 'luvit/http/' .. exports.version .. ' luvi/' .. luvi.version
```
Comment (rphillips) [inline modules/http.lua]: needs a space in-between the two

Comment (rphillips) [inline modules/http.lua]: should be == nil

File: modules/http.lua [hunk 3]
```diff
@@ -351,17 +371,20 @@ function ClientRequest:done(data, encoding, cb)
   end
 end
 
-function exports.request(options, onResponse)
+function exports.parse_url(options)
```
Comment (rphillips) [inline modules/http.lua]: parse_url should probably be `parse` from the nodejs docs.

File: modules/http.lua [hunk 4]
```diff
@@ -222,9 +230,21 @@ function ClientRequest:initialize(options, callback)
...
+      user_agent_found = value
+    end
     table.insert(self, header)
   end
 
+  if not user_agent then
+    user_agent = self.getDefaultUserAgent()
+  end
+
+  if user_agent ~= '' then
```
Comment (rphillips) [inline modules/http.lua]: let's always put the user-agent header

File: modules/http.lua (hunk 3 above)
Comment (rjemanuele) [inline modules/http.lua]: it is url.parse like node.  this is more of an internal helper for http to simplify string detection

File: modules/http.lua (hunk 4 above)
Comment (rjemanuele) [inline modules/http.lua]: it does always put it unless there is one in the headers provided by the user already

**Comment by creationix** [comment]
+1

**Pull request merged**